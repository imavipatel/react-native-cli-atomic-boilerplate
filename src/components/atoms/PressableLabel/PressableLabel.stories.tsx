import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { PressableLabel } from './PressableLabel';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('PressableLabel', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => (
    <PressableLabel testID="perssable-label" onPress={() => {}}>
      Label Text
    </PressableLabel>
  ));
