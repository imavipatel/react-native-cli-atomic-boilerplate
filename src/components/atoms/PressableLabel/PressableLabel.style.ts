import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    fontWeight: '600',
    fontSize: 16,
    color: 'black',
  },
});
