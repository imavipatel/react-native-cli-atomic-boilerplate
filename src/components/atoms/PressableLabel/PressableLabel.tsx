import React, { ReactNode } from 'react';
import { Pressable } from 'react-native';
import { Label } from '../Label/Label';

interface PressableLabelProps {
  children: ReactNode;
  onPress: () => void;
  testID: string;
}

export const PressableLabel: React.FC<PressableLabelProps> = props => {
  return (
    <Pressable onPress={props.onPress} testID={props.testID}>
      <Label>{props.children}</Label>
    </Pressable>
  );
};
