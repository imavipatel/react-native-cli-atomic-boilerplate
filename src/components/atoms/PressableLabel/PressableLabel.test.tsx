import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { PressableLabel } from './PressableLabel';

describe('PressableLabel', () => {
  const spyOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(
      <PressableLabel onPress={() => {}}>
        Primary PressableLabel
      </PressableLabel>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render Pressable Label', () => {
    const component = (
      <PressableLabel testID="pressable-label" onPress={spyOnPress}>
        Primary PressableLabel
      </PressableLabel>
    );
    const { getByTestId } = render(component);
    const button = getByTestId('pressable-label');
    expect(button).toBeDefined();
  });

  it('should call the provided onPress function when pressed', () => {
    const component = (
      <PressableLabel testID="pressable-label" onPress={spyOnPress}>
        Primary PressableLabel
      </PressableLabel>
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('pressable-label'), 'onPress');
    expect(spyOnPress).toHaveBeenCalledTimes(1);
  });

  it('should render text correctly', () => {
    const component = (
      <PressableLabel testID="pressable-label" onPress={spyOnPress}>
        Primary PressableLabel
      </PressableLabel>
    );
    const { getByText } = render(component);
    expect(getByText('Primary PressableLabel')).toBeDefined();
  });
});
