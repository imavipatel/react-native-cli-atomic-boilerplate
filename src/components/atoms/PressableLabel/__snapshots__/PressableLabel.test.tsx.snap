// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableLabel should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
    testID="label"
  >
    Primary PressableLabel
  </Text>
</View>
`;
